kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/ResourceState.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.kubernetes.operator;
+
+import java.time.Instant;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.SortedMap;
+import java.util.TreeMap;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import io.kroxylicious.kubernetes.api.common.Condition;
+import io.kroxylicious.kubernetes.api.common.ConditionBuilder;
+import io.kroxylicious.proxy.tag.VisibleForTesting;
+
+import static java.util.function.Function.identity;
+
+public class ResourceState {
+
+    // we are aiming for a deterministic ordering, so we order by status if observed generation and last transition time are equal
+    @VisibleForTesting
+    static final Comparator<Condition> FRESHEST_CONDITION = Comparator.comparing(Condition::getObservedGeneration, Comparator.nullsFirst(Long::compareTo))
+            .thenComparing(Condition::getLastTransitionTime, Comparator.nullsFirst(Instant::compareTo))
```

Comment:
there was a comment about transition time only being accurate to the second, now we are using Instant.toString I think depending on the JVM version it'll be accurate to milli, micro or nano. Testing `System.out.println(Instant.now());` locally I get `2025-04-07T05:13:08.973051228Z` with Java 17.0.12-tem

So I think this could be a good variable for selecting the freshest Condition

Reply:
I'm not sure its version but host dependent i.e. Linux -vs- Win IIRC windows used to have a less precise clock. Maybe that is moot for our purposes howevver

Reply:
OHH wait our reconciler/operator is not the only source of time as we use serverSideApply and thus the API server clock is in the picture as well.